.github/workflows/ci_cd.yml

```diff
@@ -78,12 +82,17 @@ jobs:
   build-library:
     name: "Build library"
     runs-on: windows-latest
-    needs: [ smoke-tests, unit-tests, doc-build ]
+    needs: [ build-wheelhouse, unit-tests, doc-build ]
+    permissions:
+      attestations: write
+      contents: read
+      id-token: write
```

Comment:
Why is there a write 'id-token' needed here? Is it because we upload artifacts or something?

Reply:
Needed for the attestation I think, it's from the migration guide again, along with the general push to make required permissions explicit in actions